src/main/kotlin/com/atlassian/performance/tools/report/api/result/StatsMeter.kt

```diff
@@ -78,38 +153,62 @@ class StatsMeter {
      * @return Map of results for each label
      * @since 2.4.0
      */
-    fun calculate(
+    private fun calculate(
         metrics: List<ActionMetric>,
         metric: UnivariateStatistic,
-        outlierTrimming: OutlierTrimming
+        actionToOutlier: Map<String, OutlierTrimming>
     ): Map<String, Duration> {
         val labels = metrics.map { it.label }.toSet()
         return labels
             .asSequence()
-            .map { label -> calculate(label, metrics, metric, outlierTrimming) }
+            .map { label -> label to calculate(label, metrics, metric, actionToOutlier[label]) }
             .toMap()
     }
 
-    private fun calculate(
-        label: String,
-        metrics: List<ActionMetric>,
-        metric: UnivariateStatistic,
-        outlierTrimming: OutlierTrimming
-    ): Pair<String, Duration> {
-        val durationData = ActionMetricsReader().read(metrics)[label] ?: DurationData.createEmptyMilliseconds()
-        val results = measure(durationData, metric, outlierTrimming)
-        return label to results
-    }
-
     fun measure(
         data: DurationData,
         metric: UnivariateStatistic,
-        outlierTrimming: OutlierTrimming
+        outlierTrimming: OutlierTrimming?
     ): Duration {
-        val measurement = outlierTrimming.measureWithoutOutliers(
+        val outlier = outlierTrimming ?: DEFAULT_TRIMMING
+        val measurement = outlier.measureWithoutOutliers(
             data = data.stats,
             metric = metric
         )
         return data.durationMapping(measurement)
     }
-}
\ No newline at end of file
+
+    /**
+     * Calculates `metric` for the list of `metrics`. Failed actions are not included in the result.
+     *
+     * @return Map of results for each label
+     * @since 2.4.0
+     */
+    @Deprecated(
+        message = "Use calculate instead.",
```

Comment:
This message does not make sense. I'm using calculate method and you are saying I should use calculate method instead.

Reply:
That is why I added 
`        replaceWith = ReplaceWith(
            expression = "calculate(metrics, metric, trimmingPerType)"
         )`
to indicate different one. I can change a message to "Use different calculate method, this one provides only one outlierTrimming for each action." is it more clearly?